src/include/sof/lib/pm_runtime.h

```diff
@@ -47,8 +47,30 @@ enum pm_runtime_context {
 struct pm_runtime_data {
 	spinlock_t lock;	/**< lock mechanism */
 	void *platform_data;	/**< platform specific data */
+#if CONFIG_DSP_RESIDENCY_COUNTERS
+	struct r_counters_data *r_counters; /**< diagnostic DSP residency counters */
+#endif
 };
 
+#if CONFIG_DSP_RESIDENCY_COUNTERS
+/**
+ * \brief DSP residency counters
+ * R0, R1, R2 are DSP residency counters which can be used differently
+ * based on platform implementation
+ */
+enum dsp_r_state {
+	r0_r_state = 0,	/**< high performance mode */
```

Comment:
How about adding some general explanation right after "based on platform design" above, like "R0 is the highest power consumption state while R2 is the lowest power consumption state. See platform specific pm_runtime.h for the platform HW specific mapping."

Then, the cavs/pm_runtime.h would contain detailed description of **RO and LPS use in R-states.

Reply:
ok, added description to cavs/pm_runtime.h